ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/Invoker.java

```diff
@@ -163,6 +241,168 @@ protected PackageCompilation compile(Project project) throws InvokerException {
+
+            // Get class and method references
+            Class<?> clazz = classLoader.loadClass(className);
+            Method method = clazz.getDeclaredMethod(methodName, Strand.class);
+            Function<Object[], Object> methodInvocation = createInvokerCallback(method);
+
+            // Schedule and run the function and return if result is valid
+            BFuture out = scheduler.schedule(new Object[1], methodInvocation, null, null, null,
+                    PredefinedTypes.TYPE_ERROR, null, null);
+            scheduler.start();
+
+            Object result = out.getResult();
+            Throwable panic = out.getPanic();
+            addDiagnostic(Diagnostic.debug("Panic: " + panic));
+            addDiagnostic(Diagnostic.debug("Result: " + result));
+
+            if (panic != null) {
+                throw new InvokerException(panic);
+            }
+            return result;
+        } catch (ClassNotFoundException e) {
+            addDiagnostic(Diagnostic.error(className + " class not found: " + e.getMessage()));
+            throw new InvokerException(e);
+        } catch (NoSuchMethodException e) {
+            addDiagnostic(Diagnostic.error(methodName + " method not found: " + e.getMessage()));
+            throw new InvokerException(e);
+        } catch (RuntimeException e) {
+            addDiagnostic(Diagnostic.error("Unexpected error: " + e.getMessage()));
+            throw new InvokerException(e);
+        }
+    }
+
+    /**
+     * Invokes a method that is in the given class.
+     * This is directly invoked without scheduling.
+     * The method must be a static method accepting the given parameters.
+     *
+     * @param classLoader Class loader to find the class.
+     * @param className   Class name with the method.
+     * @param methodName  Method name to invoke.
+     * @param argTypes    Types of arguments.
+     * @param args        Arguments to provide.
+     * @return The result of the invocation.
+     * @throws InvokerException If invocation failed.
+     */
+    protected Object invokeDirectMethod(ClassLoader classLoader, String className, String methodName,
+                                        Class<?>[] argTypes, Object[] args) throws InvokerException {
+        try {
+            // Get class and method references
+            addDiagnostic(Diagnostic.debug(String.format("Running %s.%s directly", className, methodName)));
+            Class<?> clazz = classLoader.loadClass(className);
+            Method method = clazz.getDeclaredMethod(methodName, argTypes);
+            Object result = method.invoke(null, args);
+            addDiagnostic(Diagnostic.debug("Result: " + result));
+            return result;
+        } catch (ClassNotFoundException e) {
+            addDiagnostic(Diagnostic.error(className + " class not found: " + e.getMessage()));
+            throw new InvokerException(e);
```

Comment:
We can move this to a method and pass the required string value as a parameter. This can reduce code duplication in multiple places.